app/controllers/api/v1/classifications_controller.rb

```diff
@@ -36,10 +36,23 @@ def build_resource_for_update(update_params)
     classification
   end
 
+  def sms_ids_from_links(link_params)
+    sms = if ids = link_params.delete(:set_member_subjects)
+      SetMemberSubject.select(:id).find(ids)
+    elsif ids = link_params.delete(:subjects)
+      SetMemberSubject.joins(:subject_set)
+        .where(subject_sets: {workflow_id: link_params[:workflow]})
+        .select(:id)
+        .find_by(subject_id: ids)
```

Comment:
Perhaps use pluck if we're just interested in the ids to avoid converting the DB records to AR models.

``` ruby
SetMemberSubject.joins(:subject_set)
   .where(subject_sets: {workflow_id: link_params[:workflow]})
   .where(subject_id: ids)
   .pluck(:id)
```